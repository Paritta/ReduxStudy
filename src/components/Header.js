import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import oc from "open-color";

const propTypes = {
    onShowModal: PropTypes.func,
    onHideModal: PropTypes.func,
};

const defaultTypes = {
    onShowModal() {},
    onHideModal() {},
};

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

const Header = ({ onShowModal, onHideModal }) => {
    return (
        <div>
            <Button
                onClick={() => onShowModal({modalType: 'MODAL_REGISTER'})}>
                등록하기
            </Button>
        </div>
    )
};

Header.propTypes = propTypes;
Header.defaultTypes = defaultTypes;

export default Header;
