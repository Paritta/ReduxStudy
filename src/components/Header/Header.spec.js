import React from "react";
import { Header } from "./Header";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";

it("Card render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Header />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
