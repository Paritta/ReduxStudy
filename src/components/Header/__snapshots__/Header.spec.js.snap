// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render correctly 1`] = `
.c0 {
  text-align: right;
  padding-right: 70px;
}

.c1 {
  background: white;
  font-weight: 500;
  font-size: 0.9em;
  border: 2px solid white;
  margin: 15px 0 15px 20px;
  text-align: left;
}

.c1:hover {
  color: #adb5bd;
}

.c2 {
  background: white;
  font-weight: 500;
  font-size: 0.9em;
  border: 2px solid white;
  margin: 15px 0 15px 20px;
}

.c2:hover {
  color: #adb5bd;
}

<div>
  <div
    className="c0"
  >
    <span>
      <button
        className="c1"
      >
        About
      </button>
      <button
        className="c2"
        onClick={[Function]}
      >
        회원가입
      </button>
      <button
        className="c2"
        onClick={[Function]}
      >
        로그인
      </button>
    </span>
  </div>
</div>
`;
